app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentModel.kt

```diff
@@ -31,11 +32,16 @@ data class ScheduleAppointmentModel(
         nextButtonState: NextButtonState
     ): ScheduleAppointmentModel {
       val potentialAppointmentDates = generatePotentialAppointmentDatesForScheduling(timeToAppointments, userClock)
+      val potientialAppointmentDateModel = PotentialAppointmentDateModel(
```

Comment:
`scheduleAppointmentDate` will not be always null, default appointment date will be loaded.

Reply:
Yes, it won't be done here right? When you are creating a new instance of `PotentialAppointmentDateModel` in this function it will always be `null` right?

Reply:
Yes it will, but that is going to be null when we add `PotentialAppointmentDateModel` from the factory method as well, right? Can you tell me what is the difference?